src/main.rs

```diff
@@ -146,7 +146,15 @@ async fn handle_get_criteria(
 ) -> Result<Response, (StatusCode, String)> {
     let mut criteria_groups: CriteriaGroups = CriteriaGroups::new(); // this is going to be aggregated criteria for all the sites
 
-    for site in query.clone().sites {
+    let mut sites = query.clone().sites;
```

Comment:
I don't think you need the clone here you can just take ownership of sites and declare it as mutable

Reply:
OK changed